lib/footer.js

```diff
@@ -3,6 +3,7 @@
  * @typedef {import('mdast').FootnoteDefinition} FootnoteDefinition
  * @typedef {import('hast').Element} Element
  * @typedef {import('hast').ElementContent} ElementContent
+ * @typedef {import('hast').Properties} Properties
```

Comment:
This line is not needed here, because I don’t think it’s used here?

Reply:
There seems to be some other problem.
Can you locally follow the error message to the file it talks about, and see what goes wrong there?

Reply:
Sorry i am a bit confused here. I have rolled back the import in the footer, and tried to solve the issue, but it seems i am not looking at the right place. Do you any advice?

Reply:
Adding the following line on L16 in `lib/footer.js` fixes the error:

```js
 * @returns {Element|null}
```

It forces a specific return type for TypeScript. Otherwise it treats the object as an optional return type, but then it doesn’t understand the spread.